packages/babel-generator/src/buffer.js

```diff
@@ -14,6 +14,7 @@ export default class Buffer {
     this._indent = format.indent.base;
     this.format = format;
     this.buf = "";
+    this.last = "";
```

Comment:
@DmitrySoshnikov flatten happens only once for each string object, see [String::SlowFlatten](https://github.com/v8/v8/blob/master/src/objects.cc#L1716)